File: java/leetcode082.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
	public ListNode deleteDuplicates(ListNode head) {
		ListNode dummy = new ListNode(0);
		ListNode fast = head, slow = dummy;
		slow.next = fast;
		while (fast != null) {
			while (fast.next != null && fast.val == fast.next.val) {
				fast = fast.next;
			}
			if (slow.next != fast) {
				slow.next = fast.next;
			} else {
				slow = slow.next;
			}
			fast = fast.next;
		}
		return dummy.next;
	}
}

File: python/leetcode082.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummy = ListNode(0)
        slow = dummy
        flag = False
        while head and head.next:
            if head.val != head.next.val:
                if not flag:
                    slow.next = head
                    slow = slow.next
                flag = False
            else:
                flag = True
            head = head.next
        slow.next = None if flag else head
        return dummy.next